rdt/transformers/id.py

```diff
@@ -164,27 +203,35 @@ def _fit(self, data):
         self.reset_randomization()
         self.data_length = len(data)
 
+        if self.cardinality_rule == 'match':
+            is_nan = int(pd.isna(data).any())  # nans count as a unique value
+            self._data_cardinality = data.nunique() + is_nan
+            self._unique_regex_values = self._generate_unique_regex_values()
```

Comment:
why do this in the fit instead of the beginning of reverse_transform? We can just do it there and store the value as an attribute for future calls to reverse transform

Reply:
Is there any reason it would need to generate during fit? I think it makes more sense to just do the actual sampling during sampling.

Reply:
Do you want to just move the `_generate_unique_regex_values` call to the first time `reverse_transform` is called? Or do you want to optimize this, e.g. if the `reverse_transform` data has fewer values than the fit data, only a subset of the regexes gets generated?

Reply:
I'm fine with either. Whichever works out to be cleaner and easier